File: JavaToKotlin/app/src/main/java/io/dustin/apps/board/domain/follow/service/ReadFollowService.java
package io.dustin.apps.board.domain.follow.service;

import io.dustin.apps.board.domain.follow.model.Follow;
import io.dustin.apps.board.domain.follow.repository.FollowRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.stream.Collectors;

@RequiredArgsConstructor
@Service
public class ReadFollowService {

    private final FollowRepository followRepository;

    public List<Follow> getFollowingIdList(Long followerId) {return followRepository.findByFollowerId(followerId);}
    public List<Long> followingIds(Long followerId) {return getFollowingIdList(followerId).stream().map(Follow::getFollowingId).collect(Collectors.toList());}

    public List<Follow> getFollowerIdList(Long followingId) {return followRepository.findByFollowingId(followingId);}
    public List<Long> followerIds(Long followingId) {return getFollowerIdList(followingId).stream().map(Follow::getFollowerId).collect(Collectors.toList());}





}

File: apps/src/main/kotlin/io/dustin/apps/board/domain/follow/service/ReadFollowService.kt
package io.dustin.apps.board.domain.follow.service

import io.dustin.apps.board.domain.follow.model.Follow
import io.dustin.apps.board.domain.follow.repository.FollowRepository
import io.dustin.apps.common.utils.notFoundEntity
import org.springframework.stereotype.Service
import java.io.Serializable

@Service
class ReadFollowService (
    private val followRepository: FollowRepository
) {

    /**
     * 팔로우 객체(단일)을 가져온다
     */
    fun getFollowObj(followerId: Long, followingId: Long): Follow? {
        return followRepository.findByFollowerIdAndFollowingId(followerId, followingId)
    }

    /**
     * 팔로우 객체(단일)을 가져온다
     */
    fun getFollow(followerId: Long, followingId: Long): Follow {
        val errorMessage = "조회된 정보가 없습니다."
        return followRepository.findByFollowerIdAndFollowingId(followerId, followingId) ?: notFoundEntity(errorMessage)
    }

    /**
     * 유저의 팔로잉리스트(팔로우객체)를 가져온다
     */
    fun getFollowingIdList(followerId: Long): List<Follow> =
        followRepository.findByFollowerId(followerId)

    /**
     * 유저의 팔로잉리스트(id)를 가져온다
     */
    fun followingIds(followerId: Long): List<Long> = getFollowingIdList(followerId).map { it.followingId }

    /**
     * 유저의 팔로워리스트(팔로우객체)를 가져온다
     */
    fun getFollowerIdList(followingId: Long): List<Follow> =
        followRepository.findByFollowingId(followingId)

    /**
     * 유저의 팔로워리스트(id)를 가져온다
     */
    fun followerIds(followingId: Long): List<Long> = getFollowerIdList(followingId).map { it.followerId }

}
